File: java_SDK/Samples/RecurringBillingSample.java
/**
 * 
 */
package Samples;

import java.io.IOException;
import java.math.BigDecimal;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

import com.payhub.ws.api.RecurringBillResponseInformation;
import com.payhub.ws.api.TransactionManager;
import com.payhub.ws.model.Bill;
import com.payhub.ws.model.CardData;
import com.payhub.ws.model.Customer;
import com.payhub.ws.model.Merchant;
import com.payhub.ws.model.MontlySchedule;
import com.payhub.ws.model.RecurringBill;
import com.payhub.ws.model.Schedule;
import com.payhub.ws.model.ScheduleSartAndEnd;
import com.payhub.ws.model.TransactionAmount;

/**
 * @author agustin
 *
 */
public class RecurringBillingSample {
	public void doRecurringBilling() throws IOException, ParseException
    {
        String url = "https://staging-api.payhub.com/api/v2/";
        String oauth = "bb96358e-2aa8-4c6c-8a2e-901b676e979d";
        ScheduleSartAndEnd scheduleSartAndEnd = new ScheduleSartAndEnd();
        scheduleSartAndEnd.setStartDate("2015-07-08");
        scheduleSartAndEnd.setEndDate("2016-07-08");
        scheduleSartAndEnd.setEnd_date_type("O");
        MontlySchedule montlySchedule = new MontlySchedule();
        montlySchedule.setMonthly_type("E");
        List<Integer> m = new ArrayList<Integer>();
        m.add(15);
        montlySchedule.setMonthly_each_days(m);
        Schedule schedule = new Schedule(scheduleSartAndEnd, montlySchedule);
        schedule.setSchedule_type("M");
        schedule.setBill_generation_interval(1);
        
        Merchant merchant = new Merchant();
        merchant.setOrganization_id(10127);
        merchant.setTerminal_id(215);
        
        Bill bill = new Bill();
        bill.setBase_amount(new TransactionAmount().dollars(new BigDecimal(100)));
        
        CardData card_data = new CardData();
        card_data.setCard_number( "5466410004374507");
        card_data.setCard_expiry_date("202012"); //September 2018
        card_data.setBilling_zip ("12345");
        
        Customer customer = new Customer();
        customer.setFirst_name("Joe");
        customer.setLast_name("Tester");
        customer.setPhone_number("844-217-1631");
        customer.setPhone_type("W");
        
        RecurringBill recurringBill = new RecurringBill(merchant,card_data,customer,schedule,bill);
        
        TransactionManager transaction = new TransactionManager(url, oauth, merchant);
        RecurringBillResponseInformation response = transaction.doRecurringBill(recurringBill);
        System.out.println(response.rowData);
    }
	public void findAll() throws IOException, ParseException
    {
        String url = "https://staging-api.payhub.com/api/v2/";
        String oauth = "bb96358e-2aa8-4c6c-8a2e-901b676e979d";
        Merchant merchant = new Merchant();
        merchant.setOrganization_id(10002);
        merchant.setTerminal_id(2);
        TransactionManager transaction = new TransactionManager(url, oauth, merchant);
        List<RecurringBillResponseInformation> response = transaction.getAllRecurringBillInformation();
        System.out.println(response.get(0).rowData);
    }
}

File: Payhub - SDK/c#_SDK/PayHubWS_sample_app/PayHubWS_sample_app/Samples/RecurringBillingSample.cs
using PayHubSDK.com.payhub.ws.api;
using PayHubSDK.com.payhub.ws.model;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using System.Globalization;


namespace PayHubWS
{
    class RecurringBillingSample
    {
        public void doRecurringBilling()
        {
			/* The current url, oauth_token, orgId and Terminal Id provided in this example, are only for testing purposes
			*  For development purposes you need to contact the Payhub Integration Support team. They will provide you with  *  all you need.
			*  Thanks.
			*/
			//Defining the Web Service URL
            string url = "https://sandbox-api.payhub.com/api/v2/";
            string oauth = "2a5d6a73-d294-4fba-bfba-957a4948d4a3";
           
            Merchant merchant = new Merchant();
            merchant.organization_id = 10074;
            merchant.terminal_id = 134;


            ScheduleSartAndEnd scheduleSartAndEnd = new ScheduleSartAndEnd();
            scheduleSartAndEnd.Start_date = Convert.ToDateTime("2016-07-08");
            scheduleSartAndEnd.End_date = Convert.ToDateTime("2017-07-08");
            scheduleSartAndEnd.end_date_type = "O";
            MontlySchedule montlySchedule = new MontlySchedule();
            montlySchedule.Monthly_type = "E";
            List<int> m = new List<int>();
            m.Add(15);
            montlySchedule.Monthly_each_days = m;
            Schedule schedule = new Schedule(scheduleSartAndEnd, montlySchedule);
            schedule.Schedule_type = "M";
            schedule.Bill_generation_interval = 1;

            Bill bill = new Bill();
            bill.Base_amount = (decimal)19.99m;
            CardData card_data = new CardData();            
            card_data.card_number = "4012881888818888";
            card_data.card_expiry_date = "202012"; //September 2018
            card_data.billing_zip = "60527";
            Customer customer = new Customer();
            customer.first_name = "Joe";
            customer.last_name = "Tester";
            customer.phone_number = "844-217-1631";
            customer.phone_type = "W";
            RecurringBill recurringBill = new RecurringBill(merchant,card_data,customer,schedule,bill);
            recurringBill.Merchant = merchant;
            recurringBill.Card_data = card_data;
            recurringBill.Customer = customer;
            recurringBill.Schedule = schedule;
            recurringBill.Bill = bill;
            TransactionManager transaction = new TransactionManager(url, oauth, merchant);
            RecurringBillInformation response = transaction.doRecurringBill(recurringBill);
            Console.Write(response.rowData);
        }       
        public void findByCustomerSample()
        {
			/* The current url, oauth_token, orgId and Terminal Id provided in this example, are only for testing purposes
			*  For development purposes you need to contact the Payhub Integration Support team. They will provide you with  *  all you need.
			*  Thanks.
			*/
			//Defining the Web Service URL
            string url = "https://sandbox-api.payhub.com/api/v2/";
            string oauth = "2a5d6a73-d294-4fba-bfba-957a4948d4a3";
           
            Merchant merchant = new Merchant();
            merchant.organization_id = 10074;
            merchant.terminal_id = 134;


            TransactionManager transaction = new TransactionManager(url, oauth, merchant);
            List<RecurringBillInformation> response = transaction.findRecurringBillInformationByCustomer("55");
            if (response!=null) Console.Write(response[0].rowData);
        }
        public void findAll()
        {
			/* The current url, oauth_token, orgId and Terminal Id provided in this example, are only for testing purposes
			*  For development purposes you need to contact the Payhub Integration Support team. They will provide you with  *  all you need.
			*  Thanks.
			*/
			//Defining the Web Service URL
            string url = "https://sandbox-api.payhub.com/api/v2/";
            string oauth = "2a5d6a73-d294-4fba-bfba-957a4948d4a3";
           
            Merchant merchant = new Merchant();
            merchant.organization_id = 10074;
            merchant.terminal_id = 134;


            TransactionManager transaction = new TransactionManager(url, oauth, merchant);
            List<RecurringBillInformation> response = transaction.getAllRecurringBillInformation(0);
            if (response != null) Console.Write(response[0].rowData);
        }
    }
}
